.jenkins/dsl/jobs.groovy

```diff
@@ -0,0 +1,184 @@
+    folder(KogitoConstants.KOGITO_DSL_RELEASE_FOLDER)
+    folder(releaseBranchFolder)
+    setupDeployJob(releaseBranchFolder, KogitoJobType.RELEASE)
+    setupPromoteJob(releaseBranchFolder, KogitoJobType.RELEASE)
+}
+
+/////////////////////////////////////////////////////////////////
+// Methods
+/////////////////////////////////////////////////////////////////
+
+void setupOptaplannerPrJob(String jobFolder) {
+    def jobParams = getDefaultJobParams()
+    jobParams.job.folder = jobFolder
+    jobParams.pr = [ blackListTargetBranches: ['7.x'] ]
+    KogitoJobTemplate.createPRJob(this, jobParams)
+}
+
+void setupOptawebEmployeeRosteringPrJob(String jobFolder) {
+    def jobParams = getDefaultJobParams('optaweb-employee-rostering')
+    jobParams.job.folder = jobFolder
+    jobParams.pr = [ whiteListTargetBranches: ['master'] ]
+    KogitoJobTemplate.createPRJob(this, jobParams)
+}
+
+void setupOptawebVehicleRoutingPrJob(String jobFolder) {
+    def jobParams = getDefaultJobParams('optaweb-vehicle-routing')
+    jobParams.job.folder = jobFolder
+    jobParams.pr = [ whiteListTargetBranches: ['master'] ]
+    KogitoJobTemplate.createPRJob(this, jobParams)
+}
+
+void setupDeployJob(String jobFolder, KogitoJobType jobType) {
+    def jobParams = getJobParams('optaplanner-deploy', jobFolder, 'Jenkinsfile.deploy', 'Optaplanner Deploy')
+    if (jobType == KogitoJobType.PR) {
+        jobParams.git.branch = '${GIT_BRABUILD_BRANCH_NAMENCH_NAME}'
+        jobParams.git.author = '${GIT_AUTHOR}'
+        jobParams.git.project_url = Utils.createProjectUrl("${GIT_AUTHOR_NAME}", jobParams.git.repository)
+    }
+    KogitoJobTemplate.createPipelineJob(this, jobParams).with {
+        parameters {
+            stringParam('DISPLAY_NAME', '', 'Setup a specific build display name')
+
+            stringParam('BUILD_BRANCH_NAME', "${GIT_BRANCH}", 'Set the Git branch to checkout')
+            if (jobType == KogitoJobType.PR) {
+                // author can be changed as param only for PR behavior, due to source branch/target, else it is considered as an env
+                stringParam('GIT_AUTHOR', "${GIT_AUTHOR_NAME}", 'Set the Git author to checkout')
+            }
+
+            booleanParam('SKIP_TESTS', false, 'Skip tests')
+
+            // Release information
+            stringParam('PROJECT_VERSION', '', 'Optional if not RELEASE. If RELEASE, cannot be empty.')
+            stringParam('KOGITO_VERSION', '', 'Optional if not RELEASE. If RELEASE, cannot be empty.')
+
+            if (jobType == KogitoJobType.PR) {
+                stringParam('PR_TARGET_BRANCH', '', 'What is the target branch of the PR?')
+            }
+        }
+
+        environmentVariables {
+            env('RELEASE', jobType == KogitoJobType.RELEASE)
+            env('JENKINS_EMAIL_CREDS_ID', "${JENKINS_EMAIL_CREDS_ID}")
+            env('MAVEN_SETTINGS_CONFIG_FILE_ID', "${MAVEN_SETTINGS_FILE_ID}")
+            
+            if (jobType == KogitoJobType.PR) {
```

Comment:
Why do we check if we are creating a PR job inside a method `setupDeployJob` ?

Reply:
The deploy job can be used in 3 cases:
- For a Nightly (KogitoType.NIGHTLY)
- For a Release (KogitoType.RELEASE)
- For a full test PR, means all artifacts repositories + images/operator (KogitoType.PR)

Each of those types generate slightly different env for the deploy job.

In the case of `KogitoType.PR`, we need to have a different env for maven deployment, as well as being able to set the `GIT_AUTHOR` as parameter (due to source author/branch)

Reply:
Makes sense; the type == PR was a bit misleading for me. Thanks!